int32 sunLight
string quality
